hangups/http_utils.py

```diff
@@ -4,61 +4,170 @@
 import asyncio
 import collections
 import logging
-import os
+import urllib.parse
 
-from hangups import exceptions
+from hangups import exceptions, channel
 
 logger = logging.getLogger(__name__)
 CONNECT_TIMEOUT = 30
 REQUEST_TIMEOUT = 30
 MAX_RETRIES = 3
 
-FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body',
-                                                         'cookies'])
+FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body'])
 
 
-@asyncio.coroutine
-def fetch(session, method, url, params=None, headers=None, data=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
+    """Session to file http requests
 
-    If the request times out or a encounters a connection issue, it will be
-    retried MAX_RETRIES times before finally raising hangups.NetworkError.
-
-    Returns FetchResponse.
+    Args:
+        cookies: dict, initial cookies for authentication
+        proxy: string, proxy url used for the request
     """
-    logger.debug('Sending request %s %s:\n%r', method, url, data)
-    error_msg = None
-    for retry_num in range(MAX_RETRIES):
-        try:
-            res = yield from asyncio.wait_for(session.request(
-                method, url, params=params, headers=headers, data=data,
-                proxy=os.environ.get('HTTP_PROXY')
-            ), CONNECT_TIMEOUT)
+    def __init__(self, cookies=None, proxy=None):
+        self._proxy = proxy
+        super().__init__(cookies=cookies)
+
+    @property
+    def cookies(self):
+        """get all cookies of the session
+
+        Returns:
+            dict, cookie name as key and cookie value as data
+        """
+        return {cookie.key: cookie.value
+                for cookie in self._cookie_jar}
```

Comment:
`ClientSession` has a public attribute we can use instead of `_cookie_jar`:
http://aiohttp.readthedocs.io/en/stable/client_reference.html#aiohttp.ClientSession.cookie_jar

Reply:
solved in https://github.com/tdryer/hangups/pull/342/commits/5ca034ba03c234305235d8a6ac3f1e1e8620f1b0